lib/src/elf.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import * as fs from 'fs';
+
+/**
+ * Read chunk of data from a file descriptor into a new Buffer.
+ */
+const readFileDescriptor = function (
```

Comment:
Nit: for consistency:

```suggestion
function readFileDescriptor(
```

 I'll add a lint to enforce this.